browser/ui/webui/brave_shields/shields_panel_data_handler.cc

```diff
@@ -52,79 +51,75 @@ void ShieldsPanelDataHandler::GetSiteBlockInfo(
 }
 
 void ShieldsPanelDataHandler::SetAdBlockMode(AdBlockMode mode) {
-  auto* shields_data_ctrlr = GetActiveShieldsDataController();
-  DCHECK(shields_data_ctrlr);
+  if (active_shields_data_controller_)
+    return;
 
-  shields_data_ctrlr->SetAdBlockMode(mode);
+  active_shields_data_controller_->SetAdBlockMode(mode);
 }
 
 void ShieldsPanelDataHandler::SetFingerprintMode(FingerprintMode mode) {
-  auto* shields_data_ctrlr = GetActiveShieldsDataController();
-  DCHECK(shields_data_ctrlr);
+  if (active_shields_data_controller_)
+    return;
 
-  shields_data_ctrlr->SetFingerprintMode(mode);
+  active_shields_data_controller_->SetFingerprintMode(mode);
 }
 
 void ShieldsPanelDataHandler::SetCookieBlockMode(CookieBlockMode mode) {
-  auto* shields_data_ctrlr = GetActiveShieldsDataController();
-  DCHECK(shields_data_ctrlr);
+  if (active_shields_data_controller_)
+    return;
 
-  shields_data_ctrlr->SetCookieBlockMode(mode);
+  active_shields_data_controller_->SetCookieBlockMode(mode);
 }
 
 void ShieldsPanelDataHandler::SetIsNoScriptsEnabled(bool is_enabled) {
-  auto* shields_data_ctrlr = GetActiveShieldsDataController();
-  DCHECK(shields_data_ctrlr);
+  if (active_shields_data_controller_)
+    return;
 
-  shields_data_ctrlr->SetIsNoScriptEnabled(is_enabled);
+  active_shields_data_controller_->SetIsNoScriptEnabled(is_enabled);
 }
 
 void ShieldsPanelDataHandler::SetHTTPSEverywhereEnabled(bool is_enabled) {
-  auto* shields_data_ctrlr = GetActiveShieldsDataController();
-  DCHECK(shields_data_ctrlr);
+  if (active_shields_data_controller_)
+    return;
 
-  shields_data_ctrlr->SetIsHTTPSEverywhereEnabled(is_enabled);
+  active_shields_data_controller_->SetIsHTTPSEverywhereEnabled(is_enabled);
 }
 
 void ShieldsPanelDataHandler::SetBraveShieldsEnabled(bool is_enabled) {
-  auto* shields_data_ctrlr = GetActiveShieldsDataController();
-  DCHECK(shields_data_ctrlr);
+  if (active_shields_data_controller_)
+    return;
 
-  shields_data_ctrlr->SetBraveShieldsEnabled(is_enabled);
+  active_shields_data_controller_->SetBraveShieldsEnabled(is_enabled);
 }
 
 void ShieldsPanelDataHandler::OpenWebCompatWindow() {
-  auto* shields_data_ctrlr = GetActiveShieldsDataController();
-  DCHECK(shields_data_ctrlr);
+  if (active_shields_data_controller_)
+    return;
 
-  OpenWebcompatReporterDialog(shields_data_ctrlr->web_contents());
+  OpenWebcompatReporterDialog(active_shields_data_controller_->web_contents());
 }
 
 BraveShieldsDataController*
 ShieldsPanelDataHandler::GetActiveShieldsDataController() {
-  auto* profile = Profile::FromWebUI(webui_controller_->web_ui());
-  DCHECK(profile);
-
-  Browser* browser = chrome::FindLastActiveWithProfile(profile);
-  if (!browser)
-    return nullptr;
-
-  auto* web_contents = browser->tab_strip_model()->GetActiveWebContents();
+  auto* web_contents = browser_->tab_strip_model()->GetActiveWebContents();
 
   if (web_contents) {
     return BraveShieldsDataController::FromWebContents(web_contents);
```

Comment:
since we're checking for NOT above to match we can reverse this logic.

`if (!web_contents) return nullptr;` and always `return BraveShieldsDataController::FromWebContents(web_contents);` looks consistent